models/service_helpers.go

```diff
@@ -179,6 +182,10 @@ func AddNewService(q *reform.Querier, serviceType ServiceType, params *AddDBMSSe
 		return nil, err
 	}
 
+	if err := validators.ValidateMySQLConnectionOptions(params.Socket, params.Address, params.Port); err != nil {
```

Comment:
Should we validate it twice? in API layer and here? 
Maybe it's better to move it above to not make unnecessary DB queries.

Reply:
Fixed.